### PR TITLE
#2457 : Improve RMNodeStarter and scheduler server network interface binding strategy

### PR DESCRIPTION
**Pull request by fviale**
- The RMNodeStarter will analyse the provided resource manager url and bind to the interface which can connect to it (selecting the fastest)
- The scheduler server will bind to the interface which can connect to the internet
- improved node.ini and server.ini explanations
- added a message in the scheduler output which displays the IP address the scheduler was bound to

### CODE REVIEW

File: rm/rm-node/src/main/java/org/ow2/proactive/resourcemanager/utils/RMNodeStarter.java [hunk 1]
```diff
@@ -453,13 +435,66 @@ private void selectNetworkInterface() {
...
+     * @throws IllegalStateException if no suitable address was found
+     */
+    public static InetAddress getNetworkInterfaceFor(String url) throws URISyntaxException, IllegalStateException {
+
+        if (url != null && CentralPAPropertyRepository.PA_NET_INTERFACE.getValue() == null &&
+                CentralPAPropertyRepository.PA_NET_NETMASK.getValue() == null && CentralPAPropertyRepository.PA_HOSTNAME != null) {
+            int port;  // use default port for rmi
+            String host;
+            URI parsedUrl = new URI(url);
+            if (parsedUrl.getScheme().equals("pamr")) {
```
Comment (sonarqube-activeeon) [inline rm/rm-node/src/main/java/org/ow2/proactive/resourcemanager/utils/RMNodeStarter.java]: MAJOR Move the "pamr" string literal on the left side of this string comparison. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1132)

File: rm/rm-node/src/main/java/org/ow2/proactive/resourcemanager/utils/RMNodeStarter.java [hunk 2]
```diff
@@ -453,13 +435,66 @@ private void selectNetworkInterface() {
...
+                // in case of pamr protocol, the url does not connect server:port information
+                if (!PAMRConfig.PA_NET_ROUTER_ADDRESS.isSet()) {
+                    logger.info(PAMRConfig.PA_NET_ROUTER_ADDRESS.getName() + " is not set, trying to connect to a local router.");
+                    host = "localhost";
+                } else {
+                    host = PAMRConfig.PA_NET_ROUTER_ADDRESS.getValue();
+                }
+                if (PAMRConfig.PA_PAMR_SOCKET_FACTORY.getValue().equals("ssh")) {
```
Comment (sonarqube-activeeon) [inline rm/rm-node/src/main/java/org/ow2/proactive/resourcemanager/utils/RMNodeStarter.java]: MAJOR Move the "ssh" string literal on the left side of this string comparison. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1132)

File: rm/rm-node/src/main/java/org/ow2/proactive/resourcemanager/utils/RMNodeStarter.java [hunk 3]
```diff
@@ -453,13 +435,66 @@ private void selectNetworkInterface() {
...
+                    if (PAMRConfig.PA_PAMRSSH_REMOTE_PORT.isSet()) {
```
Comment (sonarqube-activeeon) [inline rm/rm-node/src/main/java/org/ow2/proactive/resourcemanager/utils/RMNodeStarter.java]: MAJOR Refactor this code to not nest more than 3 if/for/while/switch/try statements. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS134)

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 3 issues:
* MAJOR 3 major

Watch the comments in this conversation to review them.

**Comment by fviale** [comment]
work on #2457 delayed for now

**Pull request closed without merging**